feat: update glee config to support multiple protocols (#351)
asyncapi/glee: docs/config-file.md

}
```

> This function must return an object with protocol specific configuration which would look like this: 

Reviewer: I think we should leave it as it was. There's also the undocumented possibility to add the following configuration keys:

* GLEE_DIR
* GLEE_LIFECYCLE_DIR
* GLEE_FUNCTIONS_DIR
* ASYNCAPI_FILE_PATH

See https://github.com/asyncapi/glee/blob/master/src/lib/configs.ts#L41.

Also `cluster` is not a protocol but the configuration of the cluster mode.

```js
export default async function () {
  return {
    websocket: {},
    mqtt: {},
    cluster: {}
  }
}

```

> Here is an example of the `glee.config.js` file for refence: 

Reviewer: ```suggestion
Here is an example of a `glee.config.js` file for reference: 
```

```js
export default async function () {
  return {
    websocket: {
      server: {
        httpServer: customServer, // A custom HTTP server of your own.
        adapter: "native", // Default. Can also be 'socket.io' or a reference to a custom adapter.
        port: process.env.PORT,
      },
      client: {
        authentication: {

>           http: () => {
>             const token = process.env.TOKEN
>             return token
>           }

Reviewer: ```suggestion
          http: () => process.env.TOKEN
```

Reviewer: BTW, [WebSocket authentication is always HTTP](https://www.rfc-editor.org/rfc/rfc6455#section-4) so it makes more sense to use `token` instead of `http`.

Author: I have removed the client example as well, and will be tackling it in #319

        }
      }
    },
    cluster: {
      adapter: "redis",
      name: "cluster", // Default. Name of your cluster.
      url: "redis://localhost:6379", // Server URL used by adapter for clustering
    },
    mqtt: {
      authentication: {
        cert: () => fs.readFileSync('./cert')
        clientId: '123',
        userPassword: {
          username: 'user1',
          password: 'pass12'
        }
      }
    }
  };
}
```
Every protocol has different configuration needs so each protocol has unique configurations:

### Websockets
|Field|Description|
|--|--|
|websocket.server|Websocket server specific configuration|

> |websocet.client| Websocket client specific configuration|

Reviewer: ```suggestion
|websocket.client| Websocket client-specific configuration|
```

Can you also please move it below to group it with the other `websocket.client` configs? It's hard to read as it is.

> |wesocket.server.adapter| The Glee adapter to use for the WebSocket server. Defaults to a "native" WebSocket implementation. Other allowed values are `socket.io` (to use the [Socket.IO](https://socket.io/) Glee adapter) or a reference to a custom adapter.|

Reviewer: ```suggestion
|websocket.server.adapter| The Glee adapter to use for the WebSocket server. Defaults to a "native" WebSocket implementation. Other allowed values are `socket.io` (to use the [Socket.IO](https://socket.io/) Glee adapter) or a reference to a custom adapter.|
```

|websocket.server.httpServer|  A custom HTTP server of your own. E.g., an [Express](https://expressjs.com/en/4x/api.html) server or any object that implements the [http.Server](https://nodejs.org/api/http.html#http_class_http_server) interface.   |

> |webocket.server.port| The port to use when binding the WebSocket server. This is useful when your server is behind a proxy and the port exposed for consumption is not the same as the port your application should be bound to. Defaults to the port specified in the selected AsyncAPI server.|

Reviewer: ```suggestion
|websocket.server.port| The port to use when binding the WebSocket server. This is useful when your server is behind a proxy and the port exposed for consumption is not the same as the port your application should be bound to. Defaults to the port specified in the selected AsyncAPI server.|
```

> |websocket.client.authentication.http| The http token to send for authentication of glee application.

Reviewer: Shouldn't this allow for other types of authentication too? I sometimes use a bearer token, and other times I use an API token/key. Depending on the server, it may require me to pass a token like `Authentication: bearer [TOKEN]`, `Authentication: token [TOKEN]`, `Authentication: api-key [token]`, etc. I think being able to say how we authenticate (using a header or a query param) and customize the header value is the minimum required here.

Author: I am keeping on one kind of authentication for this PR, since this PR just focuses on updating the config file, In #319 I will add support for all the authentication methods supported for websocket.

Author: Have removed the `websocket.client` documentation for now.

### Cluster 
|Field|Description|
|--|--|
|cluster.adapter| The Glee cluster adapter to use for communication between instances. Defaults to Redis Pub/Sub ("redis"). Can be a reference to a custom adapter.|
|cluster.name|The name of the cluster. Defaults to "cluster".|
|cluster.url|The url of the server to be used by the adapter. In case of "redis" adapter, it's the url of the Redis server.|

### MQTT

|Field|Description|
|---|---|

> |mqtt.authentication| mqtt authentication configuratation|

Reviewer: ```suggestion
|mqtt.authentication| MQTT authentication configuration|
```

This function must return an object with the following shape:

```js
export default async function () {
  return {
    websocket: {},
    mqtt: {},
    cluster: {}
  }
}

```

Here is an example of a `glee.config.js` file for reference: 
```js
export default async function () {
  return {
    websocket: {
      server: {
        httpServer: customServer, // A custom HTTP server of your own.
        adapter: "native", // Default. Can also be 'socket.io' or a reference to a custom adapter.
        port: process.env.PORT,
      }
    },
    cluster: {
      adapter: "redis",
      name: "cluster", // Default. Name of your cluster.
      url: "redis://localhost:6379", // Server URL used by adapter for clustering
    },
    mqtt: {
      authentication: {

>         cert: () => fs.readFileSync('./cert')

Reviewer: I think this function should be `async`:

```suggestion
        cert: async () => fs.readFileSync('./cert')
```
